units/src/amount/unsigned.rs

```diff
@@ -239,6 +239,7 @@ impl Amount {
     ///
     /// Returns [`None`] if overflow occurred.
     #[cfg(feature = "alloc")]
+    #[deprecated(since = "0.32.0", note = "use `checked_div_by_weight_ceil` instead ")]
     pub fn checked_div_by_weight(self, rhs: Weight) -> Option<FeeRate> {
```

Comment:
Not really.  Are you suggesting something like this:

```rust
    pub fn checked_div_by_weight_floor(self, rhs: Weight) -> Option<FeeRate> {
        let fee_rate = self.checked_div_by_weight_ceil(rhs)?.checked_sub(1)?;
        Some(fee_rate)
    }
```

This fails a test though, because if the division has no remainder, then subtracting by 1 produces the wrong results.  I guess there could be another check to see if there was no remainder but that doesn't seem more terse.